### ISSUE TITLE
[BUG] GeoIP blocks unkown IP - why are there unknown IPs ?

### ISSUE THREAD

**Issue by palijn** [tobychui/zoraxy#271]
I use GeoIP in a Whitelist for my default access rule.
When my phone is on the 4G network, its IP address is blocked, however it *is* in the right country.

A quick look at the geoipv4.csv (that I understand is used by Zoraxy?) shows that this entire IP network is not referenced. (The 77.131.*.* network)
I am not sure what the appropriate action would be.
Maybe an option while defining the access rule : "Consider unknown locations whitelisted yes/no" ?

Another way would be to support ipinfo.io as a GeoIP provider. Let the user provide their token and they're good to go.

**Comment by tobychui** [comment]
Hey @palijn 

Sorry to hear that. The IP ranges are constantly changing and, in fact, I should have make it automatically update from source. 
Using external API is not quite feasible as each request will make an API request to your geoIP resolver server and soon your quota will runs out if you load a large applications with tons of resources unless we have an algorithm / module to buffer the request IP to country map. Anyway,new ideas or suggestions are welcomed.

**Comment by palijn** [comment]
Hey @tobychui , thanks.
In this case, updating the DB wouldn't change a thing : I downloaded the most recent one and the IP range is not recorded in there.
I am slowly adding entire IP ranges when my phone suddenly gets an address in a new one, but it's cumbersome. I'd prefer at least to have the option to tell zoraxy "if you don't know, don't block it".

**Comment by tobychui** [comment]
@palijn there are *no* unknown ip range from the csv file in theory.  I just checked it manually and I found that the required IP range is listed as FR
```
77.124.0.0,77.127.255.255,IL
77.128.0.0,77.136.255.255,FR <-- in this range
77.137.0.0,77.137.127.255,IL
```

I run a test case for your specific IP range and found out in fact this is a bug in the fast geoIP resolve mode
![圖片](https://github.com/user-attachments/assets/fa3461b4-b388-4959-b799-b73c09c6cfe1)

Switching it to the default (slow) mode will resolve this issue. I guess there must be some implementation issue in the trie tree that causes this edge case error. I will look into it later.
